pandas/core/arrays/array_.py

```diff
@@ -11,11 +11,14 @@
 from pandas import compat
 
 
-def array(data,         # type: Sequence[object]
-          dtype=None,   # type: Optional[Union[str, np.dtype, ExtensionDtype]]
-          copy=True,    # type: bool
-          ):
-    # type: (...) -> ExtensionArray
+if TYPE_CHECKING:
```

Comment:
why is this needed? conditional imports for type checking are not supportable

Reply:
As I understand it, we need ExtensionArray to be defined to check this function's return type. Since ExtensionArray is defined elsewhere, we need to import it during type checking. I based this code on the approach described in [PEP 484 — Runtime or type checking](https://www.python.org/dev/peps/pep-0484/#runtime-or-type-checking).

Is there  a better, supportable way to do this?

Reply:
Yea this is the same conversation as https://github.com/pandas-dev/pandas/pull/25802#discussion_r267857645

@jreback our support of 3.5 includes 3.5.0 and 3.5.1 right? This wasn't introduced until 3.5.2

Reply:
I would instead simply use the ABCExtensionArray classes which would work for this (and lots of other clases). we should maybe just do this generally. These are by-definition available with no dependencies for this exact purpose.